### PR TITLE
Jenkins file

### PR DESCRIPTION
**Pull request by ratneshraval**
Testing new jenkinsfile for PRs.

Removes duplicate status from pull request builder.

Blocks other PRs till this is merged in, since jenkins configuration is changed to use this file.

### CODE REVIEW

**Comment by ratneshraval** [comment]
jenkins retest this please.

**Comment by ratneshraval** [comment]
jenkins retest this please

**Comment by ratneshraval** [comment]
jenkins retest this please

**Comment by ratneshraval** [comment]
jenkins, retest this please

**Comment by ratneshraval** [comment]
jenkins, retest this please

**Comment by ratneshraval** [comment]
test this please

**Comment by ratneshraval** [comment]
retest this please

**Comment by ratneshraval** [comment]
retest this please

**Comment by ratneshraval** [comment]
retest this please

**Comment by ratneshraval** [comment]
retest this please

**Comment by ratneshraval** [comment]
retest this please.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    def failureColor = '#FF0000'
+
+    try {
+        emailList = EMAIL_NOTIFICATION_LIST
+    } catch (e) {
+        // Okay not to perform assignment if EMAIL_NOTIFICATION_LIST is not defined
+    }
+
+    try {
+        stage('Install dependecies') {
```
Comment (Nasir2712) [inline Jenkinsfile]: small typo for dependencies

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        }
+
+        stage('Test - Jasmine') {
+            curStage = 'karma'
+            sh 'yarn run karma-ci'
+        }
+        stage('Test - Rspec') {
+            curStage = 'rspec'
+            withEnv([
+                'BASE_CALS_API_URL=http://localhost:3001/v1',
```
Comment (Nasir2712) [inline Jenkinsfile]: Do we still need this env variable 'BASE_CALS_API_URL=http://localhost:3001/v1' ?

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            }
+        }
+        catch(e) {
+            pipelineStatus = 'FAILED'
+            currentBuild.result = 'FAILURE'
+        }
+}
+def pushToDocker()
+{
+    // stage('Publish') {
```
Comment (Nasir2712) [inline Jenkinsfile]: is this commented out code for future reference?

Comment (ratneshraval) [reply Jenkinsfile]: Yes. Will be updating it next sprint.

**Comment by ratneshraval** [comment]
@Nasir2712 all comments addressed.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env groovy
+
+DOCKER_GROUP = 'cwds'
+DOCKER_IMAGE = 'cals'
+
+def notifyGithub(String status, String description)
+{
```
Comment (Nasir2712) [inline Jenkinsfile]: I suppose this is also for future reference?

File: Jenkinsfile (hunk 2 above)
Comment (ratneshraval) [reply Jenkinsfile]: Good point, removed.

File: Jenkinsfile (hunk 4 above)
Comment (ratneshraval) [reply Jenkinsfile]: Yes

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+}
+def reports()
+{
+    try {
+            stage ('Reports') {
+                // step([$class: 'JUnitResultArchiver', testResults: '**/reports/*.xml'])
+
```
Comment (sunilb0575) [inline Jenkinsfile]: Is this commented code for future reference?

Comment (ratneshraval) [reply Jenkinsfile]: Not sure if we will need it. But I'll clean it up in future if we don't end up using it.

**Pull request merged**